code/modules/fluids/fluid_pipe_machinery.dm

```diff
@@ -0,0 +1,521 @@
+ABSTRACT_TYPE(/obj/machinery/fluid_machinery)
+/obj/machinery/fluid_machinery
+	icon = 'icons/obj/fluidpipes/fluid_pipe.dmi'
+	desc = "Does cool things to fluids."
+	processing_tier = PROCESSING_QUARTER
+	anchored = ANCHORED
+	plane = PLANE_FLOOR
+	/// What directions are valid for connections.
+	var/initialize_directions
+	var/exclusionary = FALSE
+
+/// Accepts an input network and an ideal amount of fluid to pull from network.
+/// Returns a reagents datum containing a scaled amount of fluid linear to fullness of network or null if no fluid in network. Quantized to QUANTIZATION_UNITS units.
+/obj/machinery/fluid_machinery/proc/pull_from_network(datum/flow_network/network, maximum = 100)
+	return network.reagents.remove_any_to(max(MINIMUM_REAGENT_MOVED, round(maximum * (network.reagents.total_volume / network.reagents.maximum_volume), QUANTIZATION_UNITS)), TRUE)
```

Comment:
This feels like it should maaybe have some more significant constant term to it to make it less horribly slow at low amounts. Or at least fluid pumps should, since they are actively sucking fluid from the network.